File: c++/0167-two-sum-ii-input-array-is-sorted/solution.cpp
#include <vector>

using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int> &numbers, int target) {
        int size = numbers.size();
        if (size == 2) {
            return {1, 2};
        }

        int left = 0, right = size - 1;
        int sum = numbers[left] + numbers[right];
        while (sum != target) {
            if (sum < target) {
                sum = numbers[++left] + numbers[right];
                continue;
            }

            sum = numbers[left] + numbers[--right];
        }

        return {left + 1, right + 1};
    }
};

File: java/0167-two-sum-ii-input-array-is-sorted/Solution.java
public class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int length = numbers.length;
        if (length == 2) {
            return new int[] { 1, 2 };
        }

        int left = 0, right = length - 1;
        int sum = numbers[left] + numbers[right];
        while (sum != target) {
            if (sum < target) {
                ++left;
            }

            if (sum > target) {
                --right;
            }

            sum = numbers[left] + numbers[right];
        }

        return new int[] { left + 1, right + 1 };
    }
}
